monitor/src/main.rs

```diff
@@ -51,122 +76,92 @@ async fn main() -> Result<(), Box<dyn Error>> {
-
-        let repo_name = matches
-            .get_one::<String>("repo_name")
-            .ok_or("--repo_name argument is required")?;
-
+        let repo_name = matches.get_one::<String>("repo_name").expect("--repo_name argument is required");
         Config {
             configurations: vec![SingleConfig {
-                event: matches
-                    .get_one::<String>("event")
-                    .ok_or("--event argument is required")?
-                    .to_string(),
-                repo_owner: matches
-                    .get_one::<String>("repo_owner")
-                    .ok_or("--repo_owner argument is required")?
-                    .to_string(),
+                event: matches.get_one::<String>("event").expect("--event argument is required").to_string(),
+                repo_owner: matches.get_one::<String>("repo_owner").expect("--repo_owner argument is required").to_string(),
                 repo_name: repo_name.to_string(),
-                github_token: matches
-                    .get_one::<String>("github_token")
-                    .ok_or("--github_token argument is required")?
-                    .to_string(),
+                github_token: matches.get_one::<String>("github_token").expect("--github_token argument is required").to_string(),
                 actions_path: {
-                    let path = matches
-                        .get_one::<String>("actions_path")
-                        .ok_or("--actions_path argument is required")?
-                        .to_string();
-                    
-                    // Validation de l'existence du fichier
-                    Config::exists_actions_file(&path, repo_name)?;
+                    let path = matches.get_one::<String>("actions_path").expect("--actions_path argument is required").to_string();
+                    Config::exists_actions_file(&path.clone(), &repo_name);
                     path
-                },
+                }
             }],
+            file_path: None,
         }
     };
 
     println!("-- SealCI - Config loaded !");
     println!("{:#?}", config);
+    config
+}
 
-    let mut handles = vec![];
+async fn launch_api_server(configs: Arc<RwLock<Config>>) -> std::io::Result<()> {
+    let data = web::Data::new(AppState { configs: Arc::clone(&configs) });
+    HttpServer::new(move || {
+        App::new()
+            .app_data(data.clone())
+            .route("/configurations", web::get().to(get_configurations))
+            .route("/configurations", web::post().to(add_configuration))
+            .route("/configurations/{id}", web::get().to(get_configuration_by_id))
+            .route("/configurations/{id}", web::put().to(update_configuration))
+            .route("/configurations/{id}", web::delete().to(delete_configuration))
+            .route("/configurations/{id}/actions-file", web::get().to(get_actions_file))
+    })
+        .bind("127.0.0.1:8080")?
```

Comment:
You could put the server address as a props in configuration

Reply:
Should be fixed